docs/notebooks/demo/demo-utilities-lib/dgan_tuner_utils.py

```diff
@@ -5,21 +5,71 @@
 from gretel_client.tuner import BaseTunerMetric, MetricDirection
 
 
-# Function to pad sequences with '[END]'
+def calculate_sample_length(desired_max_len=0):
+  """Calculates the sample length and adjusted maximum sequence length
+  based on the desired ratio.
+
+  Args:
+    target_ratio: An optional integer representing the desired ratio of 
+                  max_sequence_len to sample_len. Defaults to 15, but can be
+                  overridden.
+    desired_max_len: An optional integer representing the desired maximum sequence length. 
+                      If 0, only the target_ratio is used to calculate sample_length. 
+                      If provided, it is adjusted to be divisible by target_ratio 
+                      and then used to calculate sample_length.
+
+  Returns:
+    A tuple containing two integers:
```

Comment:
nit: I see tuples with three integers being returned, looks like documentation for the last value is missing.